src/Query/Query.php

```diff
@@ -550,7 +550,7 @@ public function offset(Expression|int|null $offset): QueryInterface
         return $this;
     }
 
-    public function one(): ?array
+    public function one(): array|object|null
```

Comment:
You can add tests where each return of the method is checked, thanks.

Reply:
array and null tested [here](https://github.com/yiisoft/db/blob/7c4679af22b044567e5e35bb3a04893dac5298fd/src/TestSupport/TestQueryTrait.php#L502)

object need for active-record and tested in `yiisoft/active-record` package

Reply:
Returning in an object method because another package needs it is an error, it is conditioning the package to a dependency that it does not have, db does not depend on AR, in addition to adding a nonexistent return.

Reply:
please change return type to ?array
in ActiveRecord will be changed interface